kumascript/src/info.js

```diff
@@ -162,48 +68,81 @@ class AllPagesInfo {
-    if (!this.pagesByUri.has(uriKey)) {
-      throw new Error(`${this.getDescription(url)} does not exist`);
-    }
-    return this.pagesByUri.get(uriKey).translations;
-  }
+    // function buildTranslationObjects(data) {
+    //   // Builds a list of translation objects suitable for
+    //   // consumption by Kumascript macros, using the translation
+    //   // information from the given "data" as well as the "pageInfoByUri".
+    //   const result = [];
+    //   let rawTranslations = data.translations || [];
+    //   if (!rawTranslations.length && data.translation_of) {
+    //     const englishUri = `/en-US/docs/${data.translation_of}`;
+    //     const englishData = pageInfoByUri.get(englishUri.toLowerCase());
+    //     if (englishData) {
+    //       // First, add the English translation for this non-English locale.
+    //       result.push(
+    //         Object.freeze({
+    //           url: englishUri,
+    //           locale: "en-US",
+    //           title: englishData.title,
+    //           summary: englishData.summary,
+    //         })
+    //       );
+    //       rawTranslations = englishData.translations || [];
+    //     }
+    //   }
+    //   for (const { locale, slug } of rawTranslations) {
+    //     if (locale !== data.locale) {
+    //       // A locale is never a translation of itself.
+    //       const uri = `/${locale}/docs/${slug}`;
+    //       const pageData = pageInfoByUri.get(uri.toLowerCase());
+    //       result.push(
+    //         Object.freeze({
+    //           url: uri,
+    //           locale: locale,
+    //           title: pageData.title,
+    //           summary: pageData.summary,
+    //         })
+    //       );
+    //     }
+    //   }
+    //   return result;
+    // }
+
+    return [];
+  },
 
   getPage(url) {
-    const uriKey = this.getUriKey(url);
-    if (!this.pagesByUri.has(uriKey)) {
-      // Many of the macros expect an empty object
-      // when the requested document doesn't exist.
-      return {};
+    const result = Document.findByURL(url, { metadata: true });
+    if (!result) {
+      throw new Error(`${this.getDescription(url)} does not exist`);
     }
-    return this.pagesByUri.get(uriKey);
-  }
-
-  cacheResult(uri, result) {
-    const uriKey = this.getUriKey(uri);
-    this.resultCache.set(uriKey, result);
-  }
-
-  getResultFromCache(uri) {
-    const uriKey = this.getUriKey(uri);
-    return this.resultCache.get(uriKey);
-  }
-
-  clearCache() {
-    this.resultCache.clear();
-  }
-}
 
-module.exports = AllPagesInfo;
+    const { document } = result;
+    const { locale, slug, title, summary, tags } = document.metadata;
+    return {
+      url: document.url,
+      locale,
+      slug,
+      title,
+      summary,
+      tags: tags || [],
+      translations: [], //TODO Object.freeze(buildTranslationObjects(data)),
+      get subpages() {
+        return Document.findChildren(url, { metadata: true }).map((document) =>
```

Comment:
Probably better to use `document.url` here since it has already been "cleaned":
```suggestion
        return Document.findChildren(document.url, { metadata: true }).map((document) =>
```